dask/bytes/core.py

```diff
@@ -112,13 +112,31 @@ def read_bytes(
                     "Backing filesystem couldn't determine file size, cannot "
                     "do chunked reads. To read, set blocksize=None."
                 )
-            off = list(range(0, size, blocksize))
-            length = [blocksize] * len(off)
-            if not_zero:
-                off[0] = 1
-                length[0] -= 1
-            offsets.append(off)
-            lengths.append(length)
+
+            if size:
```

Comment:
Should this if statement be joined to the clause above? 
```python
if size is None:
    raise ValueError(...
else:
   # shrink blocksize to give same number of parts
    ...
```

Reply:
Covered below